File: java/128. Longest Consecutive Sequence/Solution.java
class Solution {
    public int longestConsecutive(int[] nums) {
        Set<Integer> numbers = new HashSet<>();
        for (int number : nums) numbers.add(number);

        int maxLength = 0;
        for (int number : numbers) {
            if (numbers.contains(number - 1)) continue;

            int current = number;
            int count = 1;

            while (numbers.contains(current + 1)) {
                current += 1;
                count++;
            }

            maxLength = Math.max(maxLength, count);
        }
        return maxLength;
    }
}

File: python/128. Longest Consecutive Sequence/solution.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        numbers = set(nums)

        longestSeq = 0
        for number in numbers:
            if number - 1 in numbers:
                continue
            current = number
            count = 1

            while current + 1 in numbers:
                current += 1
                count += 1

            longestSeq = max(longestSeq, count)

        return longestSeq